server/src/main/java/io/crate/planner/operators/NestedLoopJoin.java

```diff
@@ -225,9 +230,20 @@ public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep)
             SymbolVisitors.intersection(outputToKeep, lhs.outputs(), lhsToKeep::add);
             SymbolVisitors.intersection(outputToKeep, rhs.outputs(), rhsToKeep::add);
         }
+
         if (joinCondition != null) {
-            SymbolVisitors.intersection(joinCondition, lhs.outputs(), lhsToKeep::add);
-            SymbolVisitors.intersection(joinCondition, rhs.outputs(), rhsToKeep::add);
+            // If the outputs come only from one side and we have a lookup-join in place
+            // we can drop the join and return only the lookup-side
+            if (lhsToKeep.isEmpty() && rhsIsLookup) {
+                SymbolVisitors.intersection(joinCondition, rhs.outputs(), rhsToKeep::add);
+                return rhs.pruneOutputsExcept(rhsToKeep);
+            } else if (rhsToKeep.isEmpty() && lhsIsLookup) {
```

Comment:
If lhsIsLookup/rhsIsLookup cannot be both true at the same time, would it make sense to have a `LookupJoin` enum with `LEFT` / `RIGHT` / `NONE` ?